tests/test_target_parameters.py

```diff
@@ -314,38 +330,75 @@ def test_targeting_module_and_targeting_param_equivalent(self):
             # LoRA outputs should be the same
             assert torch.allclose(out_lora_0, out_lora_1, atol=atol, rtol=rtol)
 
-    def test_target_multiple_parameters_on_same_module(self):
-        # for now, it is not supported to target multiple parameters from the same module with the same adapter,
-        # however, it is possible to target multiple parameters from same module with different adapters
+    def test_target_multiple_parameters_on_same_module(self, monkeypatch):
+        # test that if we target multiple nn.Parameters on the same module, all of them are being used during the
+        # forward pass
         torch.manual_seed(0)
-        model_id = "hf-internal-testing/tiny-random-LlamaForCausalLM"
+        model_id = "trl-internal-testing/tiny-Llama4ForCausalLM"
         with hub_online_once(model_id):
-            model = AutoModelForCausalLM.from_pretrained(model_id)
             x = torch.arange(10).view(2, 5)
-            with torch.inference_mode():
-                out_base = model(x, output_hidden_states=True).hidden_states[-1]
+            model = MyAutoModelForCausalLM.from_pretrained(model_id)
+            shape_gate_up_proj = model.model.layers[0].feed_forward.experts.gate_up_proj.shape
+            shape_down_proj = model.model.layers[0].feed_forward.experts.down_proj.shape
+            num_layers = len(model.model.layers)
 
-            # targeting gate_up_proj
-            config0 = LoraConfig(target_parameters=["feed_forward.experts.gate_up_proj"], init_lora_weights=False)
-            model = get_peft_model(model, config0)
-            with torch.inference_mode():
-                out_lora_0 = model(x, output_hidden_states=True).hidden_states[-1]
-            atol, rtol = 1e-6, 1e-6
-            assert not torch.allclose(out_base, out_lora_0, atol=atol, rtol=rtol)
+            target_parameters = ["feed_forward.experts.gate_up_proj", "feed_forward.experts.down_proj"]
+            num_params = len(target_parameters)
+            config = LoraConfig(target_parameters=target_parameters, init_lora_weights=False)
+            model = get_peft_model(model, config)
 
-            # targeting down_proj
-            config1 = LoraConfig(target_parameters=["feed_forward.experts.down_proj"], init_lora_weights=False)
-            model.add_adapter("other", config1)
-            model.set_adapter("other")
-            with torch.inference_mode():
-                out_lora_1 = model(x, output_hidden_states=True).hidden_states[-1]
-            assert not torch.allclose(out_base, out_lora_1, atol=atol, rtol=rtol)
-            assert not torch.allclose(out_lora_0, out_lora_1, atol=atol, rtol=rtol)
+            # CHECK FORWARD CALLS
+
+            # log the weights seen during the forward call
+            weights = []
+
+            def mock_forward(self, W):
+                weights.append(W)
+                with torch.nn.utils.parametrize.cached():
+                    return W + self.delta_weight
```

Comment:
this leaks the implementation, wouldn't it be better to call the original implementation from the mock like a decorator?

Reply:
It is better, but I was lazy :-p

Fixed now.